functions/__sf_section_time.fish

```diff
@@ -0,0 +1,55 @@
+#
+# Time
+#
+
+function __sf_section_time -d "Display the current time!"
+	# ------------------------------------------------------------------------------
+	# Configuration
+	# ------------------------------------------------------------------------------
+
+	__sf_util_set_default SPACEFISH_TIME_SHOW false
+	__sf_util_set_default SPACEFISH_DATE_SHOW false
+	__sf_util_set_default SPACEFISH_TIME_PREFIX "at "
+	__sf_util_set_default SPACEFISH_TIME_SUFFIX $SPACEFISH_PROMPT_DEFAULT_SUFFIX
+	__sf_util_set_default SPACEFISH_TIME_FORMAT false
+	__sf_util_set_default SPACEFISH_TIME_12HR false
+	__sf_util_set_default SPACEFISH_TIME_COLOR "yellow"
+
+	# ------------------------------------------------------------------------------
+	# Section
+	# ------------------------------------------------------------------------------
+
+	[ $SPACEFISH_TIME_SHOW = false ]; and return
+
+	function time_date -a time_format
```

Comment:
You should be able to replace the two repeated cases with a single case with multiple arguments:
```fish
switch (uname) 
	case 'Darwin' '*BSD' 
		# MacOS + BSD use their own "date" coreutil.
		set time_format "-u" "-r 1536116421" "$time_format"
	case '*'
		set time_format "-u" "-d @1536116421" "$time_format"
end
```

Reply:
I personally installed `coreutils` on mac to get the GNU utils instead of the existing BSD ones. Maybe instead of checking uname, we should use a defining property of the util. https://stackoverflow.com/a/8748344

The following worked for me:
```fish
date --version >/dev/null
  switch $status
      case 0
          command date "-u" "-d @1536116421"
      case '*'
          command date "-u" "-r 1536116421"
  end
```

Reply:
I think I've found a slightly more elegant testing strategy for `__sf_section_time`:

Having the following function in the test file allows us to mock date with the appropriate GNU/BSD version:

```fish
function date -a time_format
	command date --version >/dev/null
	switch $status
		case 0
			command date "-u" "-d @1536116421" "$time_format"
		case '*'
			command date "-u" "-r 1536116421" "$time_format"
	end
end
```

With this, we can replace any `time_date` calls with `date`, removing any test code from the implementation 😄